File: sample/src/main/java/com/github/lany192/box/sample/ui/main/index/girl/GirlViewModel.java
package com.github.lany192.box.sample.ui.main.index.girl;

import android.os.Handler;

import androidx.lifecycle.MutableLiveData;

import com.github.lany192.arch.viewmodel.LifecycleViewModel;
import com.github.lany192.box.sample.MockUtils;
import com.github.lany192.box.sample.data.api.ApiService;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class GirlViewModel extends LifecycleViewModel {
    private final MutableLiveData<List<String>> items = new MutableLiveData<>();
    private final ApiService apiService;

    @Inject
    public GirlViewModel(ApiService apiService) {
        this.apiService = apiService;
    }

    public MutableLiveData<List<String>> getItems() {
        return items;
    }

    @Override
    protected void onLazyLoad() {
        requestCityInfo();
    }

    public void requestCityInfo() {
        showLoading(true);
        List<String> images = new ArrayList<>();
        for (int i = 0; i < 50; i++) {
            images.add(MockUtils.getImageUrl());
        }
        new Handler().postDelayed(() -> {
            showLoading(false);
            items.postValue(images);
        }, 1000);
    }

}

File: sample/src/main/java/com/github/lany192/box/sample/ui/main/index/girl/GirlViewModel.kt
package com.github.lany192.box.sample.ui.main.index.girl

import android.os.Handler
import androidx.lifecycle.MutableLiveData
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject
import com.github.lany192.box.sample.data.api.ApiService
import com.github.lany192.arch.viewmodel.LifecycleViewModel
import com.github.lany192.box.sample.MockUtils
import java.util.ArrayList

@HiltViewModel
class GirlViewModel @Inject constructor(private val apiService: ApiService) : LifecycleViewModel() {

    val items = MutableLiveData<List<String>>()

    override fun onLazyLoad() {
        requestCityInfo()
    }

    private fun requestCityInfo() {
        val images: MutableList<String> = ArrayList()
        for (i in 0..49) {
            images.add(MockUtils.getImageUrl())
        }
        Handler().postDelayed({ items.postValue(images) }, 1000)
    }
}